scripts/add_token_list.ts

```diff
@@ -0,0 +1,50 @@
+import { addTokens, getBatchExchange, getOwl } from "./util";
+import fetch from "cross-fetch";
+import { factory } from "../src/logging";
+const log = factory.getLogger("scripts.add_token_list");
+
+const argv = require("yargs")
+  .option("token_list_url", {
+    describe: "A url which can be fetched with cross-fetch",
+    default:
+      "https://raw.githubusercontent.com/gnosis/dex-js/master/src/tokenList.json",
+  })
+  .help(false)
+  .version(false).argv;
+
+interface TokenObject {
+  id: number;
+  name: string;
+  symbol: string;
+  decimals: number;
+  addressByNetwork: Record<string, string | undefined>;
+  website?: string;
+  description?: string;
+  rinkeby_faucet?: string;
+}
+
+module.exports = async function (callback: Truffle.ScriptCallback) {
+  try {
+    const tokenList: TokenObject[] = await (
+      await fetch(argv.token_list_url)
+    ).json();
+    const networkId = String(await web3.eth.net.getId());
+    const tokenAddresses = new Set(
+      tokenList.map((token) => {
+        return token.addressByNetwork[networkId] ?? "";
+      }),
+    );
+    // Ensure any non result is removed
+    tokenAddresses.delete("");
```

Comment:
nit: I prefer the implementation as it was before, is there any reason this changed?

Reply:
I was having trouble looping over the object array...

```for (const token in tokenList) { print(token) }``` was only giving me the IDs

Reply:
`for (const token of tokenList)`